### PR TITLE
Add EIP-7251 consolidation requests support

### PR DESCRIPTION
**Pull request by somnathb1**

### CODE REVIEW

File: consensus/misc/eip7251.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/consensus"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/params"
+)
+
+func DequeueConsolidationRequests7251(syscall consensus.SystemCall) types.Requests {
+	res, err := syscall(params.ConsolidationRequestAddress, nil)
+	if err != nil {
+		return nil
```
Comment (yperbasis) [inline consensus/misc/eip7251.go:13]: Plz log the error here

File: core/types/withdrawal_request.go [hunk 2]
```diff
@@ -78,8 +75,7 @@ func (s WithdrawalRequests) EncodeIndex(i int, w *bytes.Buffer) {
 	s[i].EncodeRLP(w)
 }
 
-// Requests creates a deep copy of each deposit and returns a slice of the
-// withdrwawal requests as Request objects.
+// Requests creates a deep copy of each WithdrawalRequest and returns of Requests object (as a slice).
```
Comment (yperbasis) [inline core/types/withdrawal_request.go:78]: typo in "returns of Requests object"

File: core/types/withdrawal_request.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/ledgerwatch/erigon/rlp"
 )
 
+// EIP-7251 Withdrawal Request see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7002.md
 type WithdrawalRequest struct {
```
Comment (yperbasis) [inline core/types/withdrawal_request.go:14]: "EIP-7251" shouldn't be in this comment

**Comment by somnathb1** [comment]
There are some more protocol level things in Engine API for Pectra-devnet-1 that I will add in a following PR

**Review by yperbasis — APPROVED** [review]

**Pull request merged**